dan_layer/template_macros/src/template/abi.rs

```diff
@@ -101,9 +102,20 @@ mod tests {
 
     #[test]
     fn test_hello_world() {
-        let input = TokenStream::from_str(
-            "struct HelloWorld {} impl HelloWorld { pub fn greet() -> String { \"Hello World!\".to_string() } }",
-        )
+        let input = TokenStream::from_str(indoc! {"
+            mod foo {
+                struct Foo {}
+                impl Foo {
+                    pub fn no_args_function() -> String {
+                        \"Hello World!\".to_string()
+                    }
+                    pub fn some_args_function(a: i8, b: String) -> u32 {
+                        1_u32
+                    }
+                    pub fn no_return_function() {}  
+                } 
+            }
+        "})
```

Comment:
You can also use native rust native multiline strings 

```suggestion
      let input = TokenStream::from_str(
            r#"
            mod hello_world {
                struct HelloWorld {}
                impl HelloWorld {
                    pub fn greet() -> String {
                        "Hello World!".to_string()
                    }
                } 
            }
        "#,
        )
        .unwrap();
```

Reply:
The parser was giving me a hard time with native multilines due to some span issues. Using indoc we can avoid those kind of issues and it's a dev-dependency used only in tests, so my vote goes to use indoc.

Reply:
Ah ok makes sense